components/match2/wikis/fortnite/match_group_input_custom.lua

```diff
@@ -21,31 +22,100 @@ local MatchFunctions = {
 	},
 	DEFAULT_MODE = 'team'
 }
+local FfaMatchFunctions = {}
+local MapFunctions = {}
+local FfaMapFunctions = {}
 
-local CustomMatchGroupInput = {}
+local DEFAULT_BESTOF = 3
 
 ---@param match table
 ---@param options table?
 ---@return table
 function CustomMatchGroupInput.processMatch(match, options)
-	return MatchGroupInputUtil.standardProcessFfaMatch(match, MatchFunctions)
+	return MatchGroupInputUtil.standardProcessMatch(match, MatchFunctions, FfaMatchFunctions)
+end
+
+--- Normal 2-opponent Match
+
+---@param match table
+---@param opponents table[]
+---@return table[]
+function MatchFunctions.extractMaps(match, opponents)
+	return MatchGroupInputUtil.standardProcessMaps(match, opponents, MapFunctions)
+end
+
+---@param match table
+---@param games table[]
+---@param opponents table[]
+---@return table
+function MatchFunctions.getExtraData(match, games, opponents)
+	return {
+		mvp = MatchGroupInputUtil.readMvp(match, opponents),
+		casters = MatchGroupInputUtil.readCasters(match),
+	}
+end
+
+---@param maps table[]
+---@return fun(opponentIndex: integer): integer?
+function MatchFunctions.calculateMatchScore(maps)
+	return function(opponentIndex)
+		return MatchGroupInputUtil.computeMatchScoreFromMapWinners(maps, opponentIndex)
+	end
+end
+
+---@param bestofInput string|integer?
+---@return integer
+function MatchFunctions.getBestOf(bestofInput)
```

Comment:
use the commons stuff instead (when setting up new match components that do not have any usage yet and no prerequisites from legacy match stuff always use commons)